### PR TITLE
chore: Update go devfile

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?:
_Summarize the changes. Are any stacks or samples added or updated?_

* Update schemaversion to 2.2.2
* Update endpoint protocol to https

### Which issue(s) this PR fixes:
_Link to github issue(s)_

### PR acceptance criteria:

- [ ] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [ ] Test automation
_Does this repository's tests pass with your changes?_
- [ ] Documentation
_Does any documentation need to be updated with your changes?_
- [ ] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### How to test changes / Special notes to the reviewer:

### CODE REVIEW

File: stacks/go/1.2.0/devfile.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ metadata:
   projectType: Go
   language: Go
   provider: Red Hat
-  version: 1.2.0
+  version: 1.2.1
```
Comment (thepetk) [inline stacks/go/1.2.0/devfile.yaml:12]: I think here we also have to rename the directory of `1.2.0` to be `1.2.1`

File: stacks/go/1.3.0/devfile.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ metadata:
   projectType: Go
   language: Go
   provider: Red Hat
-  version: 1.3.0
+  version: 1.3.1
```
Comment (thepetk) [inline stacks/go/1.3.0/devfile.yaml:12]: I think here we have to rename the `1.3.0` dir to `1.3.1`

File: stacks/go/stack.yaml [hunk 3]
```diff
@@ -8,13 +8,17 @@ versions:
   - version: 1.1.0
   # 1.2.0: debug command via dlv & go 1.19
```
Comment (thepetk) [inline stacks/go/stack.yaml:9]: Versions `1.2.0` & `1.3.0` should be removed

**Review by thepetk — CHANGES_REQUESTED** [review]
I've tried to test locally with `odo3` & `minikube`. I'm not able to run successfully the test, I think because of the updates to these updates:

https://github.com/devfile/registry/pull/357/files#diff-846a4935e3805e444354ad6d812de4b21939e7e2d8386f4a969f6682ab75d36fL39-L43

https://github.com/devfile/registry/pull/357/files#diff-846a4935e3805e444354ad6d812de4b21939e7e2d8386f4a969f6682ab75d36fL62-L63

I'm not sure if the `/home/user/` dir exists. We mainly use the [PROJECT_SOURCE](https://github.com/devfile/api/blob/b7100849159375a14e1960037cca21c001a75bdb/pkg/apis/workspaces/v1alpha2/commands.go#L138) as the base `workdir` for the projects.

**Comment by tolusha** [comment]
@thepetk 
I've addressed all remarks.

**Comment by thepetk** [comment]
> I've tried to test locally with `odo3` & `minikube`. I'm not able to run successfully the test, I think because of the updates to these updates:
> 
> https://github.com/devfile/registry/pull/357/files#diff-846a4935e3805e444354ad6d812de4b21939e7e2d8386f4a969f6682ab75d36fL39-L43
> 
> https://github.com/devfile/registry/pull/357/files#diff-846a4935e3805e444354ad6d812de4b21939e7e2d8386f4a969f6682ab75d36fL62-L63
> 
> I'm not sure if the `/home/user/` dir exists. We mainly use the [PROJECT_SOURCE](https://github.com/devfile/api/blob/b7100849159375a14e1960037cca21c001a75bdb/pkg/apis/workspaces/v1alpha2/commands.go#L138) as the base `workdir` for the projects.

@tolusha as mentioned in the main comment I'm not able to run the `odo dev` command using the updated devfile. I think is because of the changes made regarding the GOCACHE & GOPATH removals from the commands.

**Comment by tolusha** [comment]
@thepetk 
I got your point.
Basically I reverted everything but `https` protocol for endpoints.
I wanted to define those envs for all bash sessions, but I out of ideas.

**Review by thepetk — COMMENTED** [review]
I think it makes more readable the `devfile` if we keep the `exec.env` blocks, containing `GOPATH` & `GOCACHE` variables. So for commands `run` and `debug` I'd move the env vars in the `exec.env` and only run the required command in the `commandLine` field. WDYT?

**Comment by tolusha** [comment]
@thepetk 
I tried, but it ended up with the error:
```
 *  Executing task: devfile: build 

failed to initialize build cache at /opt/app-root/src/.cache/go-build: mkdir /opt/app-root/src/.cache: permission denied

 *  The terminal process failed to launch (exit code: 1). 
 *  Terminal will be reused by tasks, press any key to close it. 
 ```

**Comment by thepetk** [comment]
> @thepetk I tried, but it ended up with the error:
> 
> ```
>  *  Executing task: devfile: build 
> 
> failed to initialize build cache at /opt/app-root/src/.cache/go-build: mkdir /opt/app-root/src/.cache: permission denied
> 
>  *  The terminal process failed to launch (exit code: 1). 
>  *  Terminal will be reused by tasks, press any key to close it. 
> ```
> 
> ```yaml
>       env:
>         - name: GOPATH
>           value: ${PROJECT_SOURCE}/.go
>         - name: GOCACHE
>           value: ${PROJECT_SOURCE}/.cache   
> ```

Hey @tolusha ! Could you please share the devfile you've used so I can try that from my side?

**Comment by tolusha** [comment]
Sure, please try
https://raw.githubusercontent.com/tolusha/testdevfile/main/devfile.yaml

**Comment by thepetk** [comment]
> Sure, please try https://raw.githubusercontent.com/tolusha/testdevfile/main/devfile.yaml

@tolusha I wasn't able to reproduce the error with odo3. The code I've used is here:
* 1.2.1: https://github.com/thepetk/registry/blob/test/go_sample/stacks/go/1.2.1/devfile.yaml
* 1.3.1: https://github.com/thepetk/registry/blob/test/go_sample/stacks/go/1.3.1/devfile.yaml

**Comment by tolusha** [comment]
@thepetk 
Thank you for checking.
It helps me realize, that there is something wrong with command execution in eclipse-che [1]
I've moved defining env vars to env section of the command
[1] https://github.com/eclipse-che/che/issues/22930

**Comment by tolusha** [comment]
@thepetk 
1.2.0 & 1.3.0 are restored.
1.2.1 is a new default version.

**Review by thepetk — APPROVED** [review]
/lgtm

**Comment by tolusha** [comment]
@thepetk 
Could you merge the PR if everything is ok?
I don't have a merge button.

**Pull request merged**